src/main/scala/io/lambdaworks/workshop/exceptions/PasswordChecker.scala

```diff
@@ -2,19 +2,48 @@ package io.lambdaworks.workshop.exceptions
 
 object PasswordChecker {
 
-  def validate(password: String): Either[List[Throwable], String] = ???
+  def validate(password: String): Either[List[Throwable], String] = {
+    val validations : List[ Either [ Throwable, String ]] = List(
+      minNumberOfChars(password),
+      containsUpperCase(password),
+      containsLowerCase(password),
+      containsNumber(password)
+    )
 
-  private def minNumberOfChars(password: String, length: Int): Either[Throwable, String] = ???
+    val (errors, _) = validations.partition(_.isLeft)
```

Comment:
I think this step can be removed, as `validations.collect { case Left(e) => e }` should be enough.